extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -153,63 +163,103 @@ export class SubscriptionService {
    */
   fetchEntitlements_(subscriptionPlatform) {
-    subscriptionPlatform.getEntitlements().then(entitlements =>
-      this.resolveEntitlementsToStore_(entitlements));
+    subscriptionPlatform.getEntitlements().then(entitlement =>
+      this.resolveEntitlementsToStore_(entitlement));
   }
 
   /**
-   * @return {!SubscriptionService}
-   * @private
+   * @return {!Promise}
    */
-  start_() {
-    this.initialize_().then(() => {
+  start() {
+    return this.initialize_().then(() => {
+
       this.renderer_.toggleLoading(true);
 
       user().assert(this.pageConfig_, 'Page config is null');
 
-      user().assert(this.serviceConfig_['services'],
+      user().assert(this.platformConfig_['services'],
           'Services not configured in service config');
 
-      this.serviceConfig_['services'].forEach(service => {
-        this.initializeLocalPlatforms_(service,
-            /** @type {!PageConfig} */(this.pageConfig_));
-      });
-
-      const serviceIds = this.serviceConfig_['services'].map(service =>
+      const serviceIds = this.platformConfig_['services'].map(service =>
         service['serviceId'] || 'local');
 
       this.entitlementStore_ = new EntitlementStore(serviceIds);
 
+      this.platformConfig_['services'].forEach(service => {
+        this.initializeLocalPlatforms_(service);
+      });
+
       this.subscriptionPlatforms_.forEach(subscriptionPlatform => {
         this.fetchEntitlements_(subscriptionPlatform);
       });
 
       this.entitlementStore_.getGrantStatus()
           .then(grantState => {this.processGrantState_(grantState);});
 
-      this.entitlementStore_.selectPlatform()
-          .then(entitlements => {
-            this.subscriptionPlatforms_.forEach(platform => {
-              if (platform.getServiceId() == entitlements.service) {
-                platform.activate();
-              }
-            });
-          });
+      this.selectAndActivatePlatform_();
+    });
+  }
+
+  selectAndActivatePlatform_() {
+    this.entitlementStore_.selectPlatform().then(entitlement => {
+      /** @type {!RenderState} */
+      const renderState = {
+        entitlement,
+        loggedIn: false, // TODO (@prateekbh): ask how to derive this?
+        subscribed: false, // TODO (@prateekbh): ask how to derive this?
+        granted: this.grantState_,
+      };
+
+      const selectedPlatform = dev().assert(
+          this.subscriptionPlatforms_[entitlement.service],
+          'Selected service not registered'
+      );
+
+      selectedPlatform.activate(renderState);
     });
-    return this;
+  }
+
+  /**
+   * Returns Page config
+   * @returns {?PageConfig}
+   */
+  getPageConfig() {
+    return this.pageConfig_;
+  }
+
+  /**
+   * Re authorizes a platform
+   * @param {SubscriptionPlatform} subscriptionPlatform
+   */
+  reAuthorizePlatform(subscriptionPlatform) {
+    subscriptionPlatform.getEntitlements().then(() =>
+      this.selectAndActivatePlatform_());
```

Comment:
Feels like something is missing. Where does the new `getEntitlements()` response go? Shouldn't it be re-processed by the entitlements store? Also, how is grantState is updated?

It also feels like `getGrantState()` (as a promise) should belong in the entitlements tore?

Reply:
I added a TODO for implementing `reset` for this